### PR TITLE
feat(prog-check): Added UOC counts to badges and titles

### PR DESCRIPTION
**Pull request by ollibowers**
Added UOC counts to planned course badges (seen bottom right).
Also added UOC count statuses to the titles of the subgroups. 

### Concise:
image

### Full:
image

### CODE REVIEW

**Comment by ollibowers** [comment]
"Doest it behold greatness to thee?" - William Shakespeare, English Poet, 1587

File: frontend/src/pages/ProgressionChecker/GridView/GridViewSubgroup.jsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ const GridViewSubgroup = ({
 }) => {
   const { Title } = Typography;
 
+  const plannedUOC = subgroupEntries.reduce((sum, course) => (sum + (course.uoc ?? 0)), 0);
```
Comment (gtangelo) [inline frontend/src/pages/ProgressionChecker/GridView/GridViewSubgroup.jsx:15]: huh never knew of the `??` syntax before. interesting to know ahaha

File: frontend/src/pages/ProgressionChecker/CourseBadge/CourseBadge.jsx [hunk 2]
```diff
@@ -24,12 +37,19 @@ const CourseBadge = ({ course }) => {
...
 
   if (course.past) {
-    return <CourseButton course={course} planned />;
+    return (
+      <div style={{ position: "relative" }}>
+        <CourseButton course={course} planned />
+        <UOCBadge uoc={course.uoc} />
+      </div>
+    );
+    // return
```
Comment (gtangelo) [inline frontend/src/pages/ProgressionChecker/CourseBadge/CourseBadge.jsx]: remove comment

File: frontend/src/config/constants.js [hunk 3]
```diff
@@ -24,5 +24,6 @@ export const lightGrey = "#565652";
 export const darkGrey = "#323739";
 export const yellow = "#FAAD14";
 export const purple = "#9254de";
+export const lightPurple = "#efdbff";
```
Comment (gtangelo) [inline frontend/src/config/constants.js:27]: is light purple used anywhere LOL. If not remove it?

**Review by gtangelo — APPROVED** [review]
Screen Shot 2022-07-22 at 8 54 37 am

**Review by zax-xyz — APPROVED** [review]
LGTM

File: frontend/src/pages/ProgressionChecker/GridView/GridViewSubgroup.jsx (hunk 1 above)
Comment (olllllli) [reply frontend/src/pages/ProgressionChecker/GridView/GridViewSubgroup.jsx:15]: [its actually over powered](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)

File: frontend/src/config/constants.js (hunk 3 above)
Comment (olllllli) [reply frontend/src/config/constants.js:27]: it should be 😡😡😡  (term planner 👀)

**Pull request merged**